src/client/gemini_client.rs

```diff
@@ -0,0 +1,287 @@
+use bytes::Bytes;
+use nu_protocol::ShellError;
+use reqwest::Response;
+use serde::{Deserialize, Serialize};
+
+pub struct GeminiClient {
+    api_key: String,
+    max_tokens: usize,
+}
+
+// While Gemini does not have a per request limit, the per minute token limit is used here
+const MAX_FREE_TIER_TOKENS: usize = 1000000;
+
+// According to the Gemini API docs: A token is equivalent to about 4 characters for Gemini models
+const CHARS_PER_TOKEN: usize = 4;
+
+const MAX_EMBEDDING_DIMENSION: u32 = 768;
+
+// At most 100 requests can be in one batch
+const MAX_BATCH_SIZE: usize = 100;
+
+impl GeminiClient {
+    pub fn new(api_key: String, max_tokens: Option<usize>) -> Self {
```

Comment:
This just like a "this is a nice feature we could use" in future, rather than a suggestion to change:
`max_tokens: impl Into<Option<usize>>`
then `let max_tokens = max_tokens.into()`. It lets use called the function with max_tokens just set to something like 5000 rather than Some(5000)

Reply:
That's a nice idea, have made the change.